File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;

        char str[] = s.toCharArray();
        
        Stack<Character> stack = new Stack<>();
        
        if(str[0] == ')' || str[str.length - 1] == '(') answer = false;
        else {
            for(int i = 0;i < str.length;i++) {
            if(str[i] == ')') {
                if(stack.isEmpty()) {
                    answer = false;
                    break;
                }
                else
                    stack.pop();
            }
            else stack.push(str[i]);
        }
        
        if(!stack.isEmpty()) answer = false;
        }

        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    
    const input = s.split("");
    const stack = [];
    
    input.map((i) => {
        if(i === '(') stack.push(i);
        else {
            if(stack.length === 0) {
                answer = false;
                return;
            }
            else stack.pop();
        }
    })
    
    if(stack.length !== 0) answer = false;

    return answer;
}